tests/test_helpers.py

```diff
@@ -789,6 +789,22 @@ def test_is_expected_content_type_json_non_lowercase():
     )
 
 
+def test_is_expected_content_type_json_with_charset():
```

Comment:
Nit: instead of copy-pasting, I'd prefer having the same test be consolidated with `@pytest.mark.parametrize()`